test/Distributed/Runtime/distributed_actor_func_calls_remoteCall_through_generic.swift

```diff
@@ -0,0 +1,72 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend-emit-module -emit-module-path %t/FakeDistributedActorSystems.swiftmodule -module-name FakeDistributedActorSystems -disable-availability-checking %S/../Inputs/FakeDistributedActorSystems.swift
+// RUN: %target-build-swift -module-name main -Xfrontend -disable-availability-checking -j2 -parse-as-library -I %t %s %S/../Inputs/FakeDistributedActorSystems.swift -o %t/a.out
+// RUN: %target-run %t/a.out | %FileCheck %s --color --dump-input=always
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: distributed
+
+// rdar://76038845
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: back_deployment_runtime
+
+// FIXME(distributed): Distributed actors currently have some issues on windows, isRemote always returns false. rdar://82593574
+// UNSUPPORTED: OS=windows-msvc
+
+import Distributed
+import FakeDistributedActorSystems
+
+
+typealias DefaultDistributedActorSystem = FakeRoundtripActorSystem
+
+protocol DistributedWorker: DistributedActor where ActorSystem == DefaultDistributedActorSystem {
+  associatedtype WorkItem: Sendable & Codable
+  associatedtype WorkResult: Sendable & Codable
+
+  distributed func submit(work: WorkItem) async throws -> WorkResult
+}
+
+distributed actor TheWorker: DistributedWorker {
+  typealias ActorSystem = DefaultDistributedActorSystem
+  typealias WorkItem = String
+  typealias WorkResult = String
+
+  distributed func submit(work: WorkItem) async throws -> WorkResult {
+    "\(Self.self) echo: \(work)"
+  }
+}
+
+func test() async throws {
+  let system = DefaultDistributedActorSystem()
+
+  let w = TheWorker(actorSystem: system)
+  let remoteW = try! TheWorker.resolve(id: w.id, using: system)
+  print("remoteW is remote: \(__isRemoteActor(remoteW))")
+
+  // direct calls work ok:
+//  let replyDirect = try await remoteW.submit(work: "Direct")
+//  print("reply direct: \(replyDirect)")
+  // CHECK: remoteCall: on:main.TheWorker, target:main.TheWorker.submit(work:), invocation:FakeInvocationEncoder(genericSubs: [], arguments: ["Direct"], returnType: Optional(Swift.String), errorType: Optional(Swift.Error)), throwing:Swift.Error, returning:Swift.String
+  // CHECK: reply direct: TheWorker echo: Direct
+
+  print("==== ---------------------------------------------------")
+  // Fails with:
+  // Cannot handle types other than extensions and actor declarations in distributed function checking.
+  // UNREACHABLE executed at /Users/ktoso/code/swift-project/swift/lib/Sema/TypeCheckDistributed.cpp:514!
+  func callWorker<W: DistributedWorker>(w: W) async throws -> String where W.WorkItem == String, W.WorkResult == String {
+    try await w.submit(work: "Hello")
```

Comment:
The problem is that this calls the local function now, after all the making it not crash with the fixes above.

I think if we fix this to call the witness == thunk, we're good... but not sure how to...

Reply:
```
<<<<<<
          1: | assign id: ActorAddress(address: "<unique-id>") for TheWorker
          2: | actor ready: main.TheWorker
          3: | resolve ActorAddress(address: "<unique-id>") as remote // this system always resolves as remote
          4: remoteW is remote: true
          5:  > encode argument name:work, value: Direct
          6:  > encode error type: Swift.Error
          7:  > encode return type: Swift.String
          8:  > done recording
          9:  >> remoteCall: on:main.TheWorker, target:main.TheWorker.submit(work:), invocation:FakeInvocationEncoder(genericSubs: [], arguments: ["Direct"], returnType: Optional(Swift.String), errorType: Optional(Swift.Error)), throwing:Swift.Error, returning:Swift.String
         10:  > decode return type: Swift.String
         11:  > decode argument: Direct
         12:  << onReturn: TheWorker echo: Direct
         13:  << remoteCall return: TheWorker echo: Direct
         14: reply direct: TheWorker echo: Direct
check:61                                         X error: no match found
         15: ==== ---------------------------------------------------
check:61     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         16: reply: TheWorker echo: Hello
check:61     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
```